packages/mutations-ts/src/index.ts

```diff
@@ -37,6 +84,40 @@ export const createMutations = <TConfig extends ConfigSetters>(
 
   return {
     execute: async (mutationQuery: MutationQuery) => {
+
+      const metadataLink = new HttpLink({ uri: `${graphqlEndpoint}/subgraphs` });
+      const cache = new InMemoryCache();
+
+      const client = new ApolloClient({
+        cache,
+        link: metadataLink,
+        queryDeduplication: false,
+        defaultOptions: {
+          watchQuery: {
+            fetchPolicy: 'cache-and-network',
+          },
+        },
+      });
+
+      const metadata = await client.query({
+        query: GET_SUBGRAPHS
+      })
+
+      //TODO: Use Proxy to build dataSources object, reformat and make more typesafe
+
+      let dataSources = {} as any;
```

Comment:
I think that we should come up with some clever way of implementing this where we can query things on-demand as much as possible. Here's a thought:

- Create `DataSources` class
- `DataSources` has a catch-all getter (see [here](https://stackoverflow.com/questions/994143/javascript-getter-for-all-properties/36111309#36111309)) which returns a `new DataSource(target, name)` class where target === DataSources and name === datasource name the user is asking for.
- In the `DataSource` class have 2 async getters for `abi` and `address`. Whenever a getter is used, its result is stored inside the DataSources class instance (which was passed to DataSource's constructor in the above step).

Usage looks like this:
```typescript
const datasources = new DataSources(graphqlEndpoint)
const { MyContract } = datasources

await MyContract.address // fetches using datasources' query link
await MyContract.abi // fetches IPFS hash & then fetches the JSON via an IPFS request

await MyContract.abi // no fetching required, stored in datasources' cache
```